### PR TITLE
Improve basic functionality in preparation of reskin

### PR DESCRIPTION
**Pull request by knuton**
This changeset contains improvements that became necessary or advisable for adding a new set of sprites to the game.

In order:

- Dead code removal and simplification of dependencies
- Fix several bugs
- Performance improvements
- Track convenient metrics and make skier immortal

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -186,14 +186,6 @@ function startNeverEndingGame (images) {
     }
   })
 
-  game.afterCycle(function () {
```
Comment (etaque) [inline js/main.js:190]: You could also consider removing the `livesLeft` variable and `resetGame` function for complete cleanup.

File: js/lib/monster.js [hunk 2]
```diff
@@ -25,7 +23,7 @@ var Sprite = require('./sprite');
         } else {
           spriteVersion += 0.1
         }
-        if (xDiff >= 0) {
+        if (that.movingToward[0] > that.mapPosition[0]) {
           return 'sEast' + Math.ceil(spriteVersion)
         } else if (xDiff < 0) {
```
Comment (etaque) [inline js/lib/monster.js]: `xDiff` isn't set anymore, this caused the game to crash while testing:

```
Uncaught ReferenceError: xDiff is not defined
    at spritePartToUse (monster.js:28)
    at Sprite.Monster.that.draw (monster.js:33)
    at SpriteArray.<anonymous> (game.js:111)
    at S (sugar-1.3.7.min.js:27)
    at SpriteArray.each (sugar-1.3.7.min.js:37)
    at Game.that.draw (game.js:110)
    at Game.step (game.js:172)
```

**Review by etaque — CHANGES_REQUESTED** [review]
Commits are looking good, except one that is causing a crash.

File: js/lib/monster.js (hunk 2 above)
Comment (knuton) [reply js/lib/monster.js]: Of course, that should be an `else` now, thanks!

I might have mucked that up when resorting commits for a rebase.

c581966

File: js/main.js (hunk 1 above)
Comment (knuton) [reply js/main.js:190]: Yes, why not, I removed some unused things in 227aeb0.

In general, for a while I was hoping that our changes would be only some minor changes on top of the original codebase. But I think that ship has sailed.

**Review by etaque — APPROVED** [review]
Looking good :+1:

**Pull request merged**